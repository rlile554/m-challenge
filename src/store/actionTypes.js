export const REQUEST_ARTICLES = 'REQUEST_ARTICLES';
export const RECIEVE_ARTICLES = 'RECIEVE_ARTICLES';

export const SET_SEARCH = "SET_SEARCH";