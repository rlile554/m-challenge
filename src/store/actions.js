import { getArticlesByPage } from "../services/newsAPI";

const { SET_SEARCH, REQUEST_ARTICLES, RECIEVE_ARTICLES } = require("./actionTypes")

export const setSearch = (searchString) => {
	return {
		type: SET_SEARCH,
		searchString
	};
};

export const setSearchAndFetch = (searchString) => {
	return function (dispatch) {
		dispatch(setSearch(searchString));
		dispatch(fetchArticles(searchString, 1));
	}
}

export const requestArticles = () => {
	return {
		type: REQUEST_ARTICLES
	}
}

export const recieveArticles = (articles, page)  => {
	return {
		type: RECIEVE_ARTICLES,
		articles,
		page
	}
}

export function fetchArticles(searchString, page) {
  return function (dispatch) {
    dispatch(requestArticles());

		return getArticlesByPage(page, searchString)
			.then(response => {
				const articles = response.data.articles;
				console.log(articles);
				dispatch(recieveArticles(articles, page));
			});
  }
}
