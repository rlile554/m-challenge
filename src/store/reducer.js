const {
	SET_SEARCH,
	REQUEST_ARTICLES,
	RECIEVE_ARTICLES,
} = require('./actionTypes');

const initialState = {
	searchString: '',
	currentPage: 1,
	articles: [],
	loading: false,
};

export const rootReducer = (state = initialState, action) => {
	switch (action.type) {
		case SET_SEARCH: {
			return {
				...state,
				searchString: action.searchString,
				nextPage: 0,
			};
		}
		case REQUEST_ARTICLES: {
			return {
				...state,
				loading: true,
			};
		}
		case RECIEVE_ARTICLES: {
			return {
				...state,
				loading: false,
				articles: action.articles,
				currentPage: action.page
			};
		}
		default:
			return state;
	}
};
