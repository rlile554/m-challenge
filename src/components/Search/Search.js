import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {setSearchAndFetch }  from '../../store/actions';

const Search = () => {
	const savedSearchString = useSelector((state) => state.searchString);
	const [searchString, setSearchString] = useState(savedSearchString);
	const dispatch = useDispatch();

	const handleChange = (e) => {
		setSearchString(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		console.log('here');
		dispatch(setSearchAndFetch(searchString));
	}

	return (
		<form onSubmit={handleSubmit}>
			<h1>Search News:</h1>
			<input value={searchString} onChange={handleChange} />
			<button type="submit">Search</button>
		</form>
	);
};

export default Search;