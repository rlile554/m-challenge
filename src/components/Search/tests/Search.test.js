import React from 'react';
import Search from '../Search';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store/store';

test('Link changes the class when hovered', () => {
	const component = renderer.create(
		<Provider store={store}>
			<Search />
		</Provider>
	);
	let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
});
