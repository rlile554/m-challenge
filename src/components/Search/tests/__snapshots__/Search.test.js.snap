// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Search News:
  </h1>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
